src/ensembl/src/shared/components/instant-download/instant-download-gene/InstantDownloadGene.tsx

```diff
@@ -110,7 +118,15 @@ const InstantDownloadGene = (props: Props) => {
       }
     };
 
-    fetchForGene(payload);
+    const fetch = await fetchForGene(payload);
```

Comment:
Can we change this to something else like `genePromise` or `geneRequest` as naming it `fetch`, the web API`, could be a bit misleading.

Reply:
👍 yes to not naming our variables with names reserved for the web api.

But I do not understand what's happening in this function in general. What's the purpose of the new Promise on the next line?

Reply:
Thansk @imransl  I will change it. 
Loading button onclick prop is a method that returns a promise?

Reply:
An async function returns a promise by definition 😄 No need to artificially create one.

Let's discuss this PR when you are back.